app/controllers/supplejack_api/stories/moderations_controller.rb

```diff
@@ -7,9 +7,25 @@ class ModerationsController < SupplejackApplicationController
       before_action :authenticate_admin!
 
       def index
-        @user_sets = UserSet.all_public_sets
-        render json: @user_sets, each_serializer: StoriesModerationSerializer,
-               root: 'sets', adapter: :json
+        page      = index_params[:page]&.to_i        || 1
+        per_page  = index_params[:per_page]&.to_i    || 10
+        order_by  = index_params[:order_by]&.to_sym  || :updated_at
+        direction = index_params[:direction]&.to_sym || :asc
+        search    = index_params[:search]
+
+        @user_sets = UserSet.search(page, per_page, order_by, direction, search)
```

Comment:
You can pass the `index_params` to the `.search` method to avoid fat controllers. Easy to test as well.

Reply:
Do you need this to be an instance variable? `@`?